crates/sargon/src/profile/logic/profile_network/profile_network_entities_linked_to_factor_source.rs

```diff
@@ -0,0 +1,111 @@
+use crate::prelude::*;
+
+impl ProfileNetwork {
+    /// Returns the entities linked to a given `FactorSource` on the current `ProfileNetwork`.
+    pub fn entities_linked_to_factor_source(
+        &self,
+        factor_source: FactorSource,
+        integrity: FactorSourceIntegrity,
+    ) -> Result<EntitiesLinkedToFactorSource> {
+        let accounts = self
+            .accounts_non_hidden()
+            .iter()
+            .filter(|a| {
+                a.security_state
+                    .is_linked_to_factor_source(factor_source.clone())
+            })
+            .collect::<Accounts>();
+        let hidden_accounts = self
+            .accounts
+            .hidden()
```

Comment:
Minor: I guess we should have similar function to `accounts_non_hidden` on ProfileNetwork - `accounts_hidden`